core/src/main/java/org/elasticsearch/indices/recovery/RecoveryTargetService.java

```diff
@@ -263,12 +263,6 @@ public RecoveryResponse newInstance() {
                 return;
             }
 
-            if (cause instanceof IndexShardClosedException) {
```

Comment:
It's dead code. A few lines above we match the super class
`if (cause instanceof IllegalIndexShardStateException...`